modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/enricher/DynamicMetadataModelEnricher.java

```diff
@@ -96,20 +93,13 @@ private void enrichParameter(ParameterDeclaration declaration) {
     }
   }
 
-  private void enrichSourceMetadata(SourceDeclaration sourceDeclaration) {
-    final Optional<ImplementingTypeModelProperty> modelProperty =
-        sourceDeclaration.getModelProperty(ImplementingTypeModelProperty.class);
-    if (modelProperty.isPresent()) {
-      final Class<?> operationMethod = modelProperty.get().getType();
-      MetadataScopeAdapter metadataScope = new MetadataScopeAdapter(getMetadataScope(operationMethod));
-      MetadataResolverFactory metadataResolverFactory = getMetadataResolverFactory(metadataScope);
-      sourceDeclaration.setMetadataResolverFactory(metadataResolverFactory);
-
-      if (metadataScope.isCustomScope()) {
-        declareOutput(sourceDeclaration.getOutput(), sourceDeclaration.getOutputAttributes(), metadataScope);
-        declareComponentMetadataKeyId(sourceDeclaration);
-      }
-    }
+  private void enrichSourceMetadata(SourceDeclaration declaration) {
+    declaration.getModelProperty(ImplementingTypeModelProperty.class)
+        .ifPresent(prop -> {
+          final Class<?> sourceType = prop.getType();
+          MetadataScopeAdapter metadataScope = new MetadataScopeAdapter(extensionType, sourceType);
```

Comment:
IMO Sources should not be able to use this annotation, instead, as they can only declare OutputMetadata, it should be annotated with OutputResolver just like all the operations, MetadataScope should be reserved for global scenarios only